plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -165,12 +163,15 @@
 
 public final class IcebergUtil
 {
-    private static final Logger log = Logger.get(IcebergUtil.class);
-
     public static final String METADATA_FOLDER_NAME = "metadata";
     public static final String METADATA_FILE_EXTENSION = ".metadata.json";
     private static final Pattern SIMPLE_NAME = Pattern.compile("[a-z][a-z0-9]*");
     static final String TRINO_QUERY_ID_NAME = "trino_query_id";
+    // Metadata file name examples
+    //  - 00001-409702ba-4735-4645-8f14-09537cc0b2c8.metadata.json
+    //  - 00001-409702ba-4735-4645-8f14-09537cc0b2c8.gz.metadata.json (https://github.com/apache/iceberg/blob/ab398a0d5ff195f763f8c7a4358ac98fa38a8de7/core/src/main/java/org/apache/iceberg/TableMetadataParser.java#L141)
+    //  - 00001-409702ba-4735-4645-8f14-09537cc0b2c8.metadata.json.gz (https://github.com/apache/iceberg/blob/ab398a0d5ff195f763f8c7a4358ac98fa38a8de7/core/src/main/java/org/apache/iceberg/TableMetadataParser.java#L146)
+    private static final Pattern METADATA_FILE_NAME_PATTERN = Pattern.compile("(?<version>\\d+)-(?<uuid>[-a-zA-Z0-9]*)(?<compression>\\.[a-zA-Z0-9]+)?" + Pattern.quote(METADATA_FILE_EXTENSION) + "(?<compression2>\\.[a-zA-Z0-9]+)?");
```

Comment:
can uuid actually contain A-Z? all examples here are lowercase letters

Reply:
the same question applies to a compression extension

Reply:
> can uuid actually contain A-Z? all examples here are lowercase letters

that's a good point. uuid are written in hex (A-F)
i didn't want to distinguish lowercase vs uppercase

> the same question applies to a compression extension

in Iceberg codebase the compression extension is `""` or `".gz"`. i didn't want to hardcode for "gz" though